isis/src/control/apps/findfeatures/GenericTransform.cpp

```diff
@@ -120,6 +122,35 @@ cv::Point2f GenericTransform::inverse(const cv::Point2f &point) const {
   return ( dst[0] );
 }
 
+/**
+ * @brief Compute inverse with validation
+ *
+ * OpenCV may silenty return an empty matrix using the inv() method. There is a
+ * safer method that is applied here that will test if the matrix is invertable.
+ * If its not invertable and verify == true, then an exception is thrown.
+ *
+ * @author 2021-10-03 Kris J. Becker
+ *
+ * @param matrix Square matrix to invert
+ * @param verify If true will throw an exception if not invertable
+ *
+ * @return cv::Mat Inverted matrix
+ */
+cv::Mat GenericTransform::computeInverse(const cv::Mat &matrix,
+                                         const bool verify) {
+  cv::Mat inverse;
+  double result = cv::invert(matrix, inverse);
+  if ( verify == true ) {
+    if ( qFuzzyCompare(result+1.0, 0.0+1.0) ) {
```

Comment:
I actually looked at the [code](https://github.com/opencv/opencv/blob/8b4fa2605e1155bbef0d906bb1f272ec06d7796e/modules/core/src/lapack.cpp#L843-L1047) and it returns a boolean. I assumed it would be the determinant, hence the fuzzy compare. I changed it as you suggested.